app/api/api_v1/routers/family.py

```diff
@@ -78,19 +80,24 @@ async def search_family(request: Request) -> list[FamilyReadDTO]:
     :raises HTTPException: If nothing found a 404 is returned.
     :return list[FamilyDTO]: A list of matching families.
     """
-    query_params = {k: request.query_params[k] for k in request.query_params.keys()}
+    query_params: dict[str, Union[str, int]] = {
+        k: request.query_params[k] for k in request.query_params.keys()
+    }
 
     try:
         families = family_service.search(query_params)
     except ValidationError as e:
         raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=e.message)
-
-    if len(families) == 0:
+    except TimeoutError:
+        msg = "Request timed out fetching matching families. Try adjusting your query."
```

Comment:
Might be a good idea to log something here too, just so we know in the logs the timeout has occured

Reply:
Great shout!

Reply:
Addressed. Link to docs describing how we've configured this here https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-CONNECT-TIMEOUT